pkgs/os-specific/linux/nixos-rebuild/nixos-rebuild.sh

```diff
@@ -407,6 +407,16 @@ fi
 
 tmpDir=$(mktemp -t -d nixos-rebuild.XXXXXX)
 
+# macOS sets the TMPDIR to /var/folders/ab/abcdef.../A as a per-user setting
+# to improve security. However, this does not play well with SSH control paths:
+# unix_listener: path "/var/folders/ab/abcdef.../T/nixos-rebuild...." too long
+# for Unix domain socket
+# Unset TMPDIR in this case
```

Comment:
Setting it to empty is not the same as unsetting.
What is the behavior when empty? I suppose it makes the template a relative path, but it feels unsettling.

Reply:
It's the same according to `mktemp(1)`:

```
       -p DIR, --tmpdir[=DIR]
              interpret  TEMPLATE  relative  to DIR; if DIR is not specified, use $TMPDIR if set,
              else /tmp.  With this option, TEMPLATE must not be an absolute  name;  unlike  with
              -t, TEMPLATE may contain slashes, but mktemp creates only the final component
```

Reply:
I tested it, empty and unset are the same:

```
$ export TMPDIR=~
$ TMPDIR='' mktemp -t -d nixos-rebuild.XXXXXX
/tmp/nixos-rebuild.SyuD4v
$ TMPDIR= mktemp -t -d nixos-rebuild.XXXXXX
/tmp/nixos-rebuild.SURHaX
$ unset TMPDIR
$ mktemp -t -d nixos-rebuild.XXXXXX
/tmp/nixos-rebuild.PAp6SN
```